ui/lib/kv/addon/components/kv-delete-modal.js

```diff
@@ -0,0 +1,134 @@
+ *  All destroyRecord network requests are handled inside the component.
+ *
+ * <KvDeleteModal
+ *  @mode="destroy"
+ *  @path="my-secret"
+ *  @secret={{this.model.secret}}
+ *  @metadata={{this.model.metadata}}
+ * />
+ *
+ * @param {string} mode - Either: delete, destroy, undelete, or delete metadata.
+ * @param {string} path - The name of the secret.
+ * @param {object} secret - The kv/data model.
+ * @param {object} [metadata] - The kv/metadata model. It is only required when mode is "delete" or "metadata-delete".
+ */
+
+export default class KvDeleteModal extends Component {
+  @service flashMessages;
+  @service router;
+  @service store;
+  @tracked deleteType;
+  @tracked modalOpen = false;
+
+  get modalIntro() {
+    switch (this.args.mode) {
+      case 'delete':
+        return 'There are two ways to delete a version of a secret. Both delete actions can be un-deleted later. How would you like to proceed?';
+      case 'destroy':
+        return `This action will permanently destroy Version ${this.args.secret.version}
+        of the secret, and the secret data cannot be read or recovered later.`;
+      case 'delete metadata':
+        return 'This will permanently delete the metadata and versions of the secret. All version history will be removed. This cannot be undone.';
+      default:
+        return assert('mode must be one of undelete, delete, destroy, delete metadata.');
+    }
+  }
+
+  get generateDeleteRadioOptions() {
+    let isDeactivated = false;
+    if (this.args.secret.canReadMetadata) {
+      isDeactivated = this.args.metadata?.currentSecret.isDeactivated;
+    }
+
+    return [
+      {
+        key: 'delete-version',
+        label: 'Delete this version',
+        description: `This deletes Version ${this.args.secret.version} of the secret.`,
+        disabled: !this.args.secret.canDeleteVersion,
+        tooltipMessage: 'You do not have permission to delete a specific version.',
+      },
+      {
+        key: 'delete-latest-version',
+        label: 'Delete latest version',
+        description: 'This deletes the most recent version of the secret.',
+        disabled: !this.args.secret.canDeleteLatestVersion || isDeactivated,
+        tooltipMessage: isDeactivated
+          ? `The latest version of the secret is already ${this.args.metadata.secretState.state}.`
+          : 'You do not have permission to delete the latest version of this secret.',
+      },
+    ];
+  }
+
+  @action handleButtonClick(mode) {
+    // undelete is the only use case were a modal is not shown.
+    if (mode === 'undelete') {
+      this.deleteType = 'undelete-version';
+      this.save.perform();
+      return;
+    }
+    this.modalOpen = true;
+    // deleteType is the param used in the case/switch for deleteRecord on the adapter.
+    if (mode === 'destroy') {
+      this.deleteType = 'destroy-version';
+    }
+    if (mode === 'delete metadata') {
+      this.deleteType = 'delete-metadata';
+    }
```

Comment:
this logic can be removed if applying suggestion below of leveraging `@mode` to be the adapter key